portfolio/src/main/webapp/js/project-detail.js

```diff
@@ -8,26 +6,30 @@ import {Comment} from "./comments.js";
  */
 const projectId = new URLSearchParams(location.search).get("projectId");
 
-/**
- * All data about the current project.
- * @type {{id: string, name: string, description: string, detail: string[], tags: string[]}}
- */
-const project = projectsObject[projectId];
-
-console.assert(project !== undefined);
-
-/** Fill in information for this project. */
-document.querySelector("#name").innerHTML = project.name;
-document.querySelector("#description").innerHTML = project.description;
-
-/** A bullet-list of details about this project. */
-const detailListNode = document.querySelector("#detail");
-
-/** Create a new list element (detail) for each detail in the project. */
-for(let detail of project.detail){
-    const detailNode = document.createElement("li");
-    detailNode.innerHTML = detail;
-    detailListNode.appendChild(detailNode);
+/** Load data about this project from the server and populate UI items. */
+async function populateDetails() {
```

Comment:
A visual glitch here is that due to the async loading of details, the placeholder text is visible for a short period of time.  I'm no UX expert but would suggest either non-distracting placeholder content to set the layout, or a transition when pulling in the project information.

Reply:
Will add a patch for this, created #24 to track.